### ISSUE TITLE
mac-os container create error

### ISSUE THREAD

**Issue by mguzhuman** [glotcode/docker-run#2]
I am using docker-run inside a docker container. I am running container on MacOs Big Sur 11.2. I getting this error when running the code: 
```
{
  error: 'docker.container.create',
  message: 'Failed to create container: Failed while sending request: Failed deserialize response body: EOF while parsing a value at line 1 column 0'
}
```
Inside the container, I run next commands:
```
curl http://localhost:8088
{
  "name": "docker-run",
  "version": "1.1.0",
  "description": "Api for running code in transient docker containers"
}
```
Here I got error:
```
curl --header 'X-Access-Token: token' http://localhost:8088/version
{
  "error": "docker.version",
  "message": "Failed to get docker version: Failed while sending request: Failed deserialize response body: EOF while parsing a value at line 1 column 0"
}
```

```
curl --unix-socket /var/run/docker.sock http://localhost/version
{
"Platform":{"Name":"Docker Engine - Community"}, "Components":[{"Name":"Engine","Version":"20.10.2","Details"....
```

It is hello-world tests run. As you see it works just fine:

```
curl -XPOST --unix-socket /var/run/docker.sock -d '{"Image":"hello-world"}' -H 'Content-Type: application/json' http://localhost/containers/create
{"Id":"3a0e8c036fd8ebdba26c7144de2af6e3a0428a62b8e6a9107a587ac5f7f44594","Warnings":[]}
```

My env file: 

```
SERVER_LISTEN_ADDR=0.0.0.0
SERVER_LISTEN_PORT=8088
SERVER_WORKER_THREADS=10

API_ACCESS_TOKEN=token

DOCKER_UNIX_SOCKET_PATH=/var/run/docker.sock
DOCKER_UNIX_SOCKET_READ_TIMEOUT=3
DOCKER_UNIX_SOCKET_WRITE_TIMEOUT=3

DOCKER_CONTAINER_HOSTNAME=glot
DOCKER_CONTAINER_USER=glot
DOCKER_CONTAINER_MEMORY=500000000
DOCKER_CONTAINER_NETWORK_DISABLED=true
DOCKER_CONTAINER_ULIMIT_NOFILE_SOFT=90
DOCKER_CONTAINER_ULIMIT_NOFILE_HARD=100
DOCKER_CONTAINER_ULIMIT_NPROC_SOFT=3000
DOCKER_CONTAINER_ULIMIT_NPROC_HARD=3100
DOCKER_CONTAINER_CAP_DROP=MKNOD NET_RAW NET_BIND_SERVICE

RUN_MAX_EXECUTION_TIME=20
RUN_MAX_OUTPUT_SIZE=100000

RUST_LOG=debug

DEBUG_KEEP_CONTAINER=true

```

Also I have tried on Linux ( Ubuntu) the same container and it works just fine.

Also I have tried to change DOCKER_CONTAINER_HOSTNAME to various options but nothing helped.

So when I want to execute bash for an example container is creating but not running and returns and it is a main issue for now:
```
message: 'Failed to create container: Failed while sending request: Failed deserialize response body: EOF while parsing a value at line 1 column 0'
```

**Comment by mguzhuman** [comment]
@prasmussen  Hello. I conducted a small investigation and found out that when requesting to create a container, docker-run cannot deserialize the response. 
The body is sent in the request:
```
{"Hostname":"glot","User":"glot","AttachStdin":true,"AttachStdout":true,"AttachStderr":true,"Tty":false,"OpenStdin":true,"StdinOnce":true,"Image":"glot/bash:latest","NetworkDisabled":true,"HostConfig":{"Memory":1000000000,"Privileged":false,"CapAdd":[],"CapDrop":["MKNOD","NET_RAW","NET_BIND_SERVICE"],"Ulimits":[{"Name":"nofile","Soft":90,"Hard":100},{"Name":"nproc","Soft":3000,"Hard":3100}]}}
```
But when I run the command inside the container, I get the following response:
```
curl -XPOST --unix-socket /var/run/docker.sock -d '{"Hostname":"glot","User":"glot","AttachStdin":true,"AttachStdout":true,"AttachStderr":true,"Tty":false,"OpenStdin":true,"StdinOnce":true,"Image":"glot/bash:latest","NetworkDisabled":true,"HostConfig":{"Memory":500000000,"Privileged":false,"CapAdd":[],"CapDrop":["MKNOD","NET_RAW","NET_BIND_SERVICE"],"Ulimits":[{"Name":"nofile","Soft":90,"Hard":100},{"Name":"nproc","Soft":3000,"Hard":3100}]}}' -H 'Content-Type: application/json' http://localhost/containers/create
{"Id":"02ff15e721bb75c6dba5620523ffa755eae87a3de026a923480164f0e21b850e","Warnings":[]}
```

In [this](https://github.com/glotcode/docker-run/blob/main/src/docker_run/http_extra.rs#L89) part, content_length is 0

UPD. Same issue when starting container on Windows

**Comment by prasmussen** [comment]
Nice find. Maybe the Content-Length header is missing in the response.
I will look into it further this weekend.

**Comment by mguzhuman** [comment]
@prasmussen Yep. I just checked the returned header via curl:

```
curl -XPOST -i --unix-socket /var/run/docker.sock -d '{"Hostname":"glot","User":"glot","AttachStdin":true,"AttachStdout":true,"AttachStderr":true,"Tty":false,"OpenStdin":true,"StdinOnce":true,"Image":"glot/bash:latest","NetworkDisabled":true,"HostConfig":{"Memory":1000000000,"Privileged":false,"CapAdd":[],"CapDrop":["MKNOD","NET_RAW","NET_BIND_SERVICE"],"Ulimits":[{"Name":"nofile","Soft":90,"Hard":100},{"Name":"nproc","Soft":3000,"Hard":3100}]}}' -H 'Content-Type: application/json' http://localhost/containers/create

HTTP/1.1 201 Created
Api-Version: 1.41
Content-Type: application/json
Date: Thu, 11 Feb 2021 11:05:36 GMT
Docker-Experimental: false
Ostype: linux
Server: Docker/20.10.2 (linux)
Transfer-Encoding: chunked

{"Id":"b6e866a7285ea0c263fe652a2440bbc873c42d02a45c667a374192d781165166","Warnings":[]}
```

**Comment by prasmussen** [comment]
I was able to reproduce the problem when I ran docker-run inside a container.
The problem should be fixed in [1.2.0](https://github.com/glotcode/docker-run/releases/tag/v1.2.0) which supports chunked responses.

I also created the [glot/docker-run](https://hub.docker.com/r/glot/docker-run) image, check the [installation instructions](https://github.com/glotcode/docker-run/blob/main/docs/install/docker-ubuntu-20.10.md) if you are interested.

Thank you for the debugging and the detailed error reports :)

**Comment by mguzhuman** [comment]
@prasmussen Hello. Thanks. Everything is working fine now!
